features/storage/kvstore/filesystemstore/FileSystemStore.h

```diff
@@ -0,0 +1,229 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_FILE_SYSTEM_STORE_H
+#define MBED_FILE_SYSTEM_STORE_H
+
+#include "KVStore.h"
+#include "FileSystem.h"
+
+#define FSST_REVISION 1
+#define FSST_MAGIC 0x46535354 // "FSST" hex 'magic' signature
+#define FSST_PATH_NAME_SIZE 16
+#ifndef FSST_FOLDER_PATH
+#define FSST_FOLDER_PATH "$fsst$" //default FileSystemStore folder path on fs
+#endif
+
+namespace mbed {
+
+//Important data structures
+// Key metadata
+typedef struct {
+    uint32_t magic;
+    uint16_t metadata_size;
+    uint16_t revision;
+    uint32_t user_flags;
+} key_metadata_t;
+
+// incremental set handle
+typedef struct {
+    char *key;
+    uint32_t create_flags;
+    size_t data_size;
+} inc_set_handle_t;
+
+// iterator handle
+typedef struct {
+    void *dir_handle;
+    char *prefix;
+} key_iterator_handle_t;
+
+
+/** FileSystemStore for Secure Store
+ *
+ *  @code
+ *  ...
+ *  @endcode
+ */
+class FileSystemStore : KVStore {
```

Comment:
This should be a public inheritance. i.e.:
`class FileSystemStore : public KVStore {`

Reply:
Fixed